torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -618,10 +618,36 @@ void IndexCompute::handle(Merge* merge) {
   }
 }
 
+void IndexCompute::handle(Swizzle2D* swizzle_2d) {
+  auto out_x_id = maybeGetExactMapConcreteID(swizzle_2d->outX());
+  auto out_y_id = maybeGetExactMapConcreteID(swizzle_2d->outY());
+  auto in_x_id = maybeGetExactMapConcreteID(swizzle_2d->inX());
+  auto in_y_id = maybeGetExactMapConcreteID(swizzle_2d->inY());
+
+  auto out_x_it = index_map_.find(out_x_id);
+  auto out_y_it = index_map_.find(out_y_id);
+
+  if (out_x_it == index_map_.end() || out_y_it == index_map_.end()) {
+    return;
+  }
+
+  const auto out_x_ind = out_x_it->second;
+  const auto out_y_ind = out_y_it->second;
+
+  // Actual swizzle operation is handled via IndexSwizzle pass
```

Comment:
Will this be moved when the legacy swizzle is removed, or is the plan to handle swizzle in a separate pass?

Reply:
I think handling on a separate pass would be more flexible to support all the use cases more easily.

Reply:
What's the advantage of having it on a separate pass instead of simply doing it on the final traversal of the indexing?

Reply:
Actually I think they should be unified eventually so eventually will be 2 pass in total including the concrete pass, the final traversal, and the swizzle. 

Some additional clean up will be needed on the final traversal end to be able to combine with the swizzle pass though. I have follow up PRs on that soon.